### DISCUSSION TITLE
Error with selecting time of DataArray

### DISCUSSION THREAD

**Discussion by erkoj** [pydata/xarray#4924] [Q&A] [answered]
I want to analyze different time periods of an regular xarray.DataArray.
xx.resample(time='Y').min()  gives error "Replacing the dayofyr or dayofwk of a datetime is not supported".
xx.sel(time='1990-01-01') gives the same error.
This issue is in Jupyter notebook I run in my laptop (server version is 5.7.4, cftime version is 1.4.1), in Cheyenne with the same data and code everything works without problem. I feel the problem is connected with the cftime, but what exactly and how to solve it?

All the best,
Erko Jakobson

**Comment by spencerkclark** [comment, marked as answer]
Thanks @erkoj -- could you provide a minimal working example?  The latest version of cftime (1.4.1) did come with at least one issue that we had to address within xarray (#4870).  I would be very interested to know if there was another, as we are set to release a new version of xarray this weekend (#4894).

I'm having trouble reproducing your errors with the development version of xarray with a simple example I concocted, but that could mean either your issue is subtly different, or this is somehow related to using an older version of xarray with the latest cftime:

```
In [1]: import cftime; import xarray as xr

In [2]: cftime.__version__
Out[2]: '1.4.1'

In [3]: times = xr.cftime_range("1990", periods=1000)

In [4]: da = xr.DataArray(range(1000), dims=["time"], coords=[times])

In [5]: da.resample(time="Y").min()
Out[5]:
<xarray.DataArray (time: 3)>
array([  0, 365, 730])
Coordinates:
  * time     (time) object 1990-12-31 00:00:00 ... 1992-12-31 00:00:00

In [6]: da.sel(time="1990-01-01")
Out[6]:
<xarray.DataArray (time: 1)>
array([0])
Coordinates:
  * time     (time) object 1990-01-01 00:00:00
```

**Reply by erkoj** [reply]
Thank you very much - the hint about xarray version helped - I upgraded from version 0.13.0 to 0.16.2 and now everything works well.

**Reply by spencerkclark** [reply]
Awesome, indeed upon reflection I remember seeing this before.  In case anyone comes across this question, it is fixed by upgrading to at least xarray version 0.14.1 (see [this issue](https://github.com/pydata/xarray/issues/3426) and [fix](https://github.com/pydata/xarray/pull/3430) for reference).
